code/modules/modular_computers/file_system/programs/generic/folding.dm

```diff
@@ -0,0 +1,122 @@
+GLOBAL_LIST_INIT(science_strings, list(
+	"Extruding Mesh Terrain", "Virtualizing Microprocessor",
+	"Reticulating Splines", "Inserting Chaos Generator",
+	"Reversing Polarity", "Unfolding Proteins",
+	"Simulating Alien Abductions", "Scanning Pigeons",
+	"Iterating Chaos Array", "Abstracting Supermatter",
+	"Adjusting Social Network", "Recalculating Clown Principle"
+))
+
+#define MINIMUM_SCIENCE_INTERVAL 450
+#define MAXIMUM_SCIENCE_INTERVAL 900
+#define MINIMUM_FOLDING_EVENT_INTERVAL 50
+#define MAXIMUM_FOLDING_EVENT_INTERVAL 100
+#define SCIENCE_MONEY_PER_MINUTE 0.08 // So little money.
+
+/datum/computer_file/program/folding
+	filename = "fldng"
+	filedesc = "FOLDING@SAPCE"
+	extended_desc = "This program uses processor cycles for science, in exchange for money."
+	program_icon_state = "generic"
+	program_key_state = "generic_key"
+	size = 6
+	available_on_network = 1
+	usage_flags = PROGRAM_ALL
+	nanomodule_path = /datum/nano_module/program/folding
+	category = PROG_UTIL
+	requires_network = 1
+
+	var/started_on 			= 0 	// When the program started some science.
+	var/current_interval 	= 0		// How long the current interval will be.
+	var/next_event 			= 0 	// in world timeofday, when the next event is scheduled to pop.
+	var/crashed				= FALSE	// Program periodically needs a restart.
+	var/crashed_at			= 0		// When the program crashed.
+
+/datum/computer_file/program/folding/Topic(href, href_list)
+	if(..())
+		return TOPIC_HANDLED
```

Comment:
Really weird pattern that preempts base checks from closing the UI or refreshing it. Is this a conscious choice?

Reply:
Yep. It's done on a significant number of all Topic() implementations.

Reply:
Well, the typical line would be
```
. = ..()
if(.)
	return
```
Not everything uses Topic return value defines properly, but those that do should try and remain consistent about it.